app/src/main/java/fr/neamar/kiss/forwarder/ExperienceTweaks.java

```diff
@@ -346,12 +338,27 @@ private boolean isNonCompliantKeyboard() {
     }
 
     /**
-     * Should the keyboard be displayed by default?
+     * Should the keyboard be displayed when opening the launcher?
      */
     private boolean isKeyboardOnStartEnabled() {
         return prefs.getBoolean("display-keyboard", false);
     }
 
+    /**
+     * Should the keyboard be displayed when opened as an assistant?
+     */
+    private boolean isKeyboardOnAssistantStartEnabled() {
+        return prefs.getBoolean("display-keyboard-assist", false);
+    }
+
+    /**
+     * Should the keyboard be displayed?
+     */
+    private boolean shouldShowKeyboard() {
+        boolean isAssistant = mainActivity.getIntent().getAction().equalsIgnoreCase("android.intent.action.ASSIST");
+        return ((isKeyboardOnStartEnabled() && !isAssistant) || (isKeyboardOnAssistantStartEnabled() && isAssistant));
```

Comment:
isn't it better to do the conditional in the other direction, to benefit from early-exit?

`(!isAssistant && isKeyboardOnStartEnabled())`

this way, is isAssistant is true, we don't even call isKeyboardOnStartEnabled and skip reading the value

Reply:
done